src/Messenger.Client/Objects/MessengerSenderAction.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+
+namespace Messenger.Client.Objects
+{
+    public enum MessengerSenderAction
+    {
+        MarkSeen,
+        TypingOn,
+        TypingOff,
+    }
+
+
+    internal static class MessengerSenderActionExtensions
+    {
+        public static string ToJSON(this MessengerSenderAction @this)
+        {
+            switch (@this)
```

Comment:
Can we use EnumMember instead?

Reply:
Tried this actually... something didn't work about it that I quite can't remember right now.
Let me spin up a branch to reproduce what I was having issues with.

Reply:
Okay got the EnumMember approach working 😊

Trick was to use '[JsonConverter(typeof(StringEnumConverter))]' so that it would serialize using the string representation not the default int representation – regardless of what settings the consuming app has for its JSON serialization